client/src/context/useContestContext.tsx

```diff
@@ -1,28 +1,39 @@
 import { createContext, FunctionComponent, useContext, useEffect, useState } from 'react';
 import { Contest } from '../interface/Contest';
 import { useAuth } from './useAuthContext';
-import { fetchAllContestByUserId } from '../helpers/APICalls/searchContest';
+import { fetchAllContestByUserId, fetchAllContests } from '../helpers/APICalls/searchContest';
 interface IContestContext {
-  readonly allContests: Contest[];
+  allContestsByUser: Contest[] | [];
+  allContests: Contest[] | [];
 }
 
 export const ContestContext = createContext<IContestContext>({
+  allContestsByUser: [],
   allContests: [],
 });
 export const ContestProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [allContestsByUser, setAllContestsByUser] = useState<Contest[] | []>([]);
+  const [allContests, setAllContests] = useState<Contest[] | []>([]);
   const { loggedInUser } = useAuth();
   // get all contests and submission by userID
   useEffect(() => {
     if (loggedInUser) {
       const getAllContestByUserId = async () => {
         const contests = await fetchAllContestByUserId({ id: loggedInUser.id });
-        setAllContext(contests.contests);
+        setAllContestsByUser(contests?.contests || []);
```

Comment:
Maybe perform a check here (use an if clause) on the response here and accordingly update the state? :)

Reply:
Successfully implemented if clause. See implementation at
commit 03e97656998da9be9788a9e83a0d734c45c19266